controllers/mutation_reconcile_looper.go

```diff
@@ -84,7 +88,16 @@ func (m *MutationReconcileLooper) ReconcileMutationObject(ctx context.Context, c
 		if err != nil {
 			return "", fmt.Errorf("fs error: %w", err)
 		}
-		defer vfs.Cleanup(virtualFS)
+
+		defer func() {
+			if err := vfs.Cleanup(virtualFS); err != nil {
+				log.Error(err, "failed to cleanup virtual filesystem")
+			}
+		}()
+
+		if !isTar(resourceData) {
+			return "", fmt.Errorf("expected tarred directory content for configuration/localization resources, got plain text")
```

Comment:
The check makes sense. How do we surface this so that the controller can set a condition on the object with a suitable reason? Probably should be some kind of error type.

Reply:
Hmm, good question.

Reply:
I'll make something happen here. :D